Packages/Tracking Preview/Locomotion/Runtime/Scripts/Teleportation/Teleport Action/Jump Gem Teleport/JumpGem.cs

```diff
@@ -0,0 +1,630 @@
+            {
+                PinchJewel(chirality);
+            }
+        }
+
+        /// <summary>
+        /// Checks to see which hand is closest. Returning false means that there is no hand present.
+        /// </summary>
+        private bool GetClosestHand(out Chirality chirality, out Vector3 pinchPosition)
+        {
+            chirality = Chirality.Left;
+            pinchPosition = Vector3.zero;
+            Hand left = leapProvider.CurrentFrame.GetHand(Chirality.Left);
+            Hand right = leapProvider.CurrentFrame.GetHand(Chirality.Right);
+            if (left == null && right == null)
+            {
+                return false;
+            }
+
+            float leftDist = 100, rightDist = 100;
+            Vector3 leftPos = Vector3.zero, rightPos = Vector3.zero;
+            if (left != null)
+            {
+                leftPos = left.GetPredictedPinchPosition();
+                leftDist = Vector3.Distance(transform.position, leftPos);
+            }
+            if (right != null)
+            {
+                rightPos = right.GetPredictedPinchPosition();
+                rightDist = Vector3.Distance(transform.position, rightPos);
+            }
+            chirality = leftDist < rightDist ? Chirality.Left : Chirality.Right;
+            pinchPosition = leftDist < rightDist ? leftPos : rightPos;
+
+            return true;
+        }
+
+        /// <summary>
+        /// Ensure the pinching hand is within a suitable distance to the gem.
+        /// </summary>
+        private bool ValidateDistanceBetweenGem(Vector3 position)
+        {
+            return Vector3.Distance(position, _pinchItem.position) <= _scaledGemSize;
+        }
+
+        private void PinchJewel(Chirality chirality)
+        {
+            _releaseTimeCurrent = _releaseTime;
+            _pinchStartTimeCurrent = _pinchStartTime;
+            _isPinched = true;
+            _hasMovedAway = false;
+            _pinchedChirality = chirality;
+            _pinchStartPosition = _pinchItem.position;
+            _pinchStartRotation = _pinchItem.rotation;
+            _pinchItem.transform.SetParent(null);
+
+            PlaySound(_pinchClip);
+            OnPinch?.Invoke(chirality);
+        }
+
+        #endregion
+
+        #region Pinching Logic
+
+        /// <summary>
+        /// Logic flow for when the gem is reported as pinched
+        /// </summary>
+        private void PinchedGem()
+        {
+            _facingAmount = 1f;
+            _pinchedHand = leapProvider.CurrentFrame.GetHand(_pinchedChirality);
+            _pinchAmount = 1f;
+            if (_pinchedHand != null)
+            {
+                _detectedHidden = false;
+
+                if (_nullTimeCurrent > 0f)
+                {
+                    _hasPinchedEnough = false;
+                    _nullTimeCurrent -= Time.deltaTime;
+                    return;
+                }
+
+                if (_pinchDetector.SquishPercent > _requiredPinchSquishAmount)
```

Comment:
The logic here seems odd - if a user is pinching enough to have grabbed the jump gem, why does the gem itself them do another pinch check to allow the activation of the teleport ray?

Surely activation of the ray should be based soley on the distance the user has moved them gem from its idle position, rather than how much they are pinching after the initial pinch check passes? Seems prone to confusing errors (as seen in Aurora).

Reply:
agree +1 

@MattGrayUL is this something you'd be up for changing or shall we look at this?

Reply:
i'm happy to look at it when I get around to review :)